go/internal/feast/featurestore.go

```diff
@@ -180,80 +176,98 @@ func (fs *FeatureStore) GetOnlineFeatures(ctx context.Context, request *serving.
 	if err != nil {
 		return nil, err
 	}
-
+	result := make([]*FeatureVector, 0)
+	arrowMemory := memory.NewGoAllocator()
 	for _, groupRef := range groupedRefs {
 		featureData, err := fs.readFromOnlineStore(ctx, groupRef.entityKeys, groupRef.featureViewNames, groupRef.featureNames)
 		if err != nil {
 			return nil, err
 		}
-		fs.populateResponseFromFeatureData(featureData,
+
+		vectors, err := fs.transposeResponseIntoColumns(featureData,
 			groupRef,
-			onlineFeatureResponse,
 			fvs,
-			offset,
+			arrowMemory,
+			numRows,
 		)
-		offset += len(groupRef.featureNames)
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, vectors...)
 	}
 	// TODO (Ly): ODFV, skip augmentResponseWithOnDemandTransforms
-	return onlineFeatureResponse, nil
+	return result, nil
 }
 
 func (fs *FeatureStore) DestructOnlineStore() {
 	fs.onlineStore.Destruct()
 }
 
-// parseFeatures parses the kind field of a GetOnlineFeaturesRequest protobuf message
+// ParseFeatures parses the kind field of a GetOnlineFeaturesRequest protobuf message
 // and populates a Features struct with the result.
-func (fs *FeatureStore) parseFeatures(kind interface{}) (*Features, error) {
+func (fs *FeatureStore) ParseFeatures(kind interface{}) (*Features, error) {
 	if featureList, ok := kind.(*serving.GetOnlineFeaturesRequest_Features); ok {
-		return &Features{features: featureList.Features.GetVal(), featureService: nil}, nil
+		return &Features{Features: featureList.Features.GetVal(), FeatureService: nil}, nil
 	}
 	if featureServiceRequest, ok := kind.(*serving.GetOnlineFeaturesRequest_FeatureService); ok {
 		featureService, err := fs.registry.getFeatureService(fs.config.Project, featureServiceRequest.FeatureService)
 		if err != nil {
 			return nil, err
 		}
-		return &Features{features: nil, featureService: featureService}, nil
+		return &Features{Features: nil, FeatureService: featureService}, nil
 	}
 	return nil, errors.New("cannot parse kind from GetOnlineFeaturesRequest")
 }
 
 // getFeatureRefs extracts a list of feature references from a Features struct.
 func (fs *FeatureStore) getFeatureRefs(features *Features) ([]string, error) {
-	if features.featureService != nil {
+	if features.FeatureService != nil {
 		var featureViewName string
 		featureRefs := make([]string, 0)
-		for _, featureProjection := range features.featureService.projections {
+		for _, featureProjection := range features.FeatureService.projections {
 			featureViewName = featureProjection.nameToUse()
 			for _, feature := range featureProjection.features {
 				featureRefs = append(featureRefs, fmt.Sprintf("%s:%s", featureViewName, feature.name))
 			}
 		}
 		return featureRefs, nil
 	} else {
-		return features.features, nil
+		return features.Features, nil
+	}
+}
+
+func (fs *FeatureStore) ExtractFeatureRefs(kind interface{}, fullFeatureNames bool) ([]string, error) {
```

Comment:
Is there a way that we can make `kind` here a type instead of `interface{}`. Haven't looked to closely at it though.

Reply:
where is `ExtractFeatureRefs` being used?